packages/manager/apps/pci-gateway/src/pages/onboarding/OnBoardingGuard.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useAllAggregatedGateway } from '@/api/hooks/useGateway';
+
+export default function OnBoardingGuard({
+  projectId,
+  children,
+}: {
+  projectId: string;
+  children: JSX.Element;
+}): JSX.Element {
+  const [displayChildren, setDisplayChildren] = useState(false);
+  const navigate = useNavigate();
+
+  const { data: aggregatedGateways, isPending } = useAllAggregatedGateway(
+    projectId,
+  );
+
+  useEffect(() => {
+    if (!isPending) {
+      if (aggregatedGateways?.length > 0) {
+        navigate(`/pci/projects/${projectId}/gateway`);
+      } else {
+        setDisplayChildren(true);
+      }
+    }
+  }, [aggregatedGateways, isPending, navigate]);
+
+  return displayChildren ? children : null;
```

Comment:
You don't need another state and an useEffect to do this
You can import Navigate component and do something like :

```
if (isPending) {
  return null; // Or a Spinner
}

return aggregatedGateways?.length > 0 ? <Navigate to={...} /> : <>{children}</>;
```